tools/buildsys/src/manifest.rs

```diff
@@ -250,12 +250,26 @@ use std::path::{Path, PathBuf};
 pub struct Error(error::Error);
 type Result<T> = std::result::Result<T, Error>;
 
+/// The possible dependency representations in a package manifest.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(untagged)]
+enum DependencyInfo {
+    Version(String),
+    Dependency { path: Option<String> },
+}
+
 /// The nested structures here are somewhat complex, but they make it trivial
 /// to deserialize the structure we expect to find in the manifest.
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "kebab-case")]
 pub struct ManifestInfo {
     package: Package,
+    #[serde(default)]
+    build_dependencies: HashMap<String, DependencyInfo>,
+    #[serde(default)]
+    dependencies: HashMap<String, DependencyInfo>,
+    #[serde(default)]
+    dev_dependencies: HashMap<String, DependencyInfo>,
```

Comment:
Is there a reason to maintain this separation?

A nicer interface might be:
```suggestion
    #[serde(default)]
    dependencies: HashSet<Manifest>,
```

However, that would make the struct relatively expensive to construct if each `Manifest` also had its own list of dependencies, since that duplicates work for repeated dependencies like `glibc`.

A compromise might be:
```suggestion
    #[serde(default)]
    dependencies: HashSet<PathBuf>,
```

Where each dependency is a path to some other buildsys package's `Cargo.toml` - then the caller could combine all of those into a larger set and do deeper manifest inspection just once.

Reply:
> Is there a reason to maintain this separation?

This is the struct used to deserialize the Cargo.toml structure into Rust. Each of these are separate sections in Cargo.toml and container the name and `DependencyInfo` data.